apidef/oas/event_test.go

```diff
@@ -255,86 +361,167 @@ func TestEventHandlers(t *testing.T) {
 }
 
 func TestEventHandler_MarshalJSON(t *testing.T) {
-	e := EventHandler{
-		Enabled: true,
-		Trigger: event.QuotaExceeded,
-		Kind:    event.WebhookKind,
-		ID:      "random-id",
-		Name:    "test-webhook",
-		Webhook: WebhookEvent{
-			URL:            "https://webhook.site/uuid",
-			Headers:        Headers{{Name: "Auth", Value: "key"}},
-			BodyTemplate:   "/path/to/template",
-			CoolDownPeriod: ReadableDuration(time.Second * 20),
-			Method:         http.MethodPost,
-		},
+	type testCase struct {
+		title    string
+		input    EventHandler
+		expected map[string]any
 	}
 
-	data, err := json.Marshal(e)
-	assert.NoError(t, err)
-	expected := map[string]interface{}{
-		"id":      "random-id",
-		"enabled": true,
-		"trigger": "QuotaExceeded",
-		"type":    "webhook",
-		"name":    "test-webhook",
-		"url":     "https://webhook.site/uuid",
-		"headers": []interface{}{
-			map[string]interface{}{
-				"name":  "Auth",
-				"value": "key",
+	testCases := []testCase{
+		{
+			title: "should marshal webhook event handler",
+			input: EventHandler{
+				Enabled: true,
+				Trigger: event.QuotaExceeded,
+				Kind:    event.WebhookKind,
+				ID:      "random-id",
+				Name:    "test-webhook",
+				Webhook: WebhookEvent{
+					URL:            "https://webhook.site/uuid",
+					Headers:        Headers{{Name: "Auth", Value: "key"}},
+					BodyTemplate:   "/path/to/template",
+					CoolDownPeriod: ReadableDuration(time.Second * 20),
+					Method:         http.MethodPost,
+				},
+			},
+			expected: map[string]any{
+				"id":      "random-id",
+				"enabled": true,
+				"trigger": "QuotaExceeded",
+				"type":    "webhook",
+				"name":    "test-webhook",
+				"url":     "https://webhook.site/uuid",
+				"headers": []any{
+					map[string]any{
+						"name":  "Auth",
+						"value": "key",
+					},
+				},
+				"bodyTemplate":   "/path/to/template",
+				"cooldownPeriod": "20s",
+				"method":         "POST",
 			},
 		},
-		"bodyTemplate":   "/path/to/template",
-		"cooldownPeriod": "20s",
-		"method":         "POST",
+		{
+			title: "should marshal jsvm event handler",
+			input: EventHandler{
+				Enabled: true,
+				Trigger: event.QuotaExceeded,
+				Kind:    event.JSVMKind,
+				ID:      "random-id",
+				Name:    "test-custom",
+				JSVMEvent: JSVMEvent{
+					FunctionName: "myCustomEventHandler",
+					Path:         "event_handlers/session_editor.js",
+				},
+			},
+			expected: map[string]any{
+				"id":           "random-id",
+				"enabled":      true,
+				"trigger":      "QuotaExceeded",
+				"type":         "custom",
+				"name":         "test-custom",
+				"functionName": "myCustomEventHandler",
+				"path":         "event_handlers/session_editor.js",
+				"body":         "console.log('hello world');",
```

Comment:
May need body next to path to be usable (mentioned on data model).

Reply:
we dropped the body part for this ticket